src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -642,7 +648,16 @@ ImmutableArray<byte> computeContentHash()
                             sourceIndex: index, destination: charBuffer,
                             destinationIndex: 0, count: charsToCopy);
 
-                        hash.Append(MemoryMarshal.AsBytes(charBuffer.AsSpan(0, charsToCopy)));
+                        var charSpan = charBuffer.AsSpan(0, charsToCopy);
+
+                        // Ensure everything is always little endian, so we get the same results across all platforms.
+                        if (isBigEndian)
+                        {
+                            var shortSpan = MemoryMarshal.Cast<char, short>(charSpan);
+                            ReverseEndianness(source: shortSpan, destination: shortSpan);
```

Comment:
I assume it's ok that this is mutating charSpan/charBuffer and that it's not data that's expected to be readonly?

Reply:
correct.  it was data we copied into this scratch buffer, purely for the pursposes of passing to xxhash64 (see earlier in the method)